src/less/chart-legend/chart-legend.less

```diff
@@ -0,0 +1,85 @@
+.chart-legend {
+    width: 100%;
+}
+
+.chart-legend dl {
+    display: flex;
+    flex-direction: column;
+    margin: 0;
+    padding: 0;
+    row-gap: var(--spacing-200);
+}
+
+.chart-legend__item {
+    align-items: center;
+    display: flex;
+    font-size: var(--font-size-small);
+    width: 100%;
+
+    &::before {
+        border: 1px solid;
+        border-radius: 2px;
+        box-sizing: border-box;
+        content: "";
+        display: inline-block;
+        height: 10px;
+        margin-inline-end: var(--spacing-100);
+        width: 10px;
+    }
+
+    &.chart-legend__item--primary {
+        &::before {
+            background-color: var(--color-data-viz-chart-primary);
+            border-color: var(--color-data-viz-chart-primary);
+        }
+    }
+
+    &.chart-legend__item--secondary {
+        &::before {
+            background-color: var(--color-data-viz-chart-secondary);
+            border-color: var(--color-data-viz-chart-secondary);
+        }
+    }
+
+    &.chart-legend__item--tertiary {
+        &::before {
+            background: repeating-linear-gradient(
+                30deg,
+                var(--color-data-viz-chart-tertiary-background),
+                var(--color-data-viz-chart-tertiary-background) 2px,
+                var(--color-data-viz-chart-tertiary-stroke) 2px,
+                var(--color-data-viz-chart-tertiary-stroke) 3px,
+                var(--color-data-viz-chart-tertiary-background) 3px,
+                var(--color-data-viz-chart-tertiary-background) 4px
+            );
+            border: 1px solid var(--color-data-viz-chart-tertiary-stroke);
+        }
+    }
+
+    &.chart-legend__item--quaternary {
+        &::before {
+            background: repeating-linear-gradient(
+                0deg,
+                var(--color-data-viz-chart-quaternary-background),
+                var(--color-data-viz-chart-quaternary-background) 2px,
+                var(--color-data-viz-chart-quaternary-stroke) 2px,
+                var(--color-data-viz-chart-quaternary-stroke) 3px,
+                var(--color-data-viz-chart-quaternary-background) 3px,
+                var(--color-data-viz-chart-quaternary-background) 4px
+            );
+            border: 1px solid var(--color-data-viz-chart-quaternary-stroke);
+        }
+    }
+
+    &.chart-legend__item--quinary {
+        &::before {
+            background-color: var(--color-data-viz-chart-quinary-background);
+            border-color: var(--color-data-viz-chart-quinary-stroke);
+        }
+    }
+}
+
+.chart-legend__value {
```

Comment:
Were the values really spec-ed to be right aligned? Per UX principles, these should probably be left-aligned to improve reading flow, consistency, scalability, and legibility. I believe it's also an `a11y` improvement for people with dyslexia.

Reply:
I'm not sure I quite follow

I would consider this to be "tabular" format, in which numeric values are usually right-aligned

Here is the comp, it's basically emulating what a `space-between` would give if this was `flex`

<img width="370" alt="image" src="https://github.com/eBay/skin/assets/3231913/5d085fc7-29c1-4c5f-808e-7d3c7df19d94">

Either case, yes to RTL support

Reply:
I think comps are sometimes "too perfect" in a sense that they relay the most optimistic visual scenario. When presented with the most pessimistic scenario, often design guidance changes. One of the only time right alignment is preferable on the web is for numeric data inside tables and there's a reason for that - tables usually have gridlines, which already creates visual separation and balance, which makes for good readability. Virtually all other text alignment (with minor exceptions) on the web is most optimal left-aligned. This is a general UX principle. See this for more info: https://beyondpixelss.medium.com/the-power-of-left-alignment-a-user-centric-perspective-in-design-8546ea330c71#

We may want to reach out to DS to get a revision to the specs.

Reply:
I definitely agree that left-aligned text is generally more readable. I think in this context, since it's a value-pair, and not a large body of text, it doesn't seem to be too much of any issue. Maybe this is something we can discuss in a larger group.

For now, I've removed the `text-align: right`, and moved the flex to the label in order to achieve the styling (this removes the need for extra `rtl` specific styling